woocommerce/woocommerce-ios
# Blaze: Add error handling and unit test for fetching payment method on payment screen

Part of #11702 


## Description

This PR adds a simple error handling for when fetching payment info screen fails. Unit tests for `BlazeConfirmPaymentViewModel` have also been added.

Also: to improve the experience, I updated the external URL handling to open the URLs in a Safari sheet instead of an external browser.

## Testing instructions

- Update `BlazeConfirmPaymentViewModel`'s method `updatePaymentInfo` to force the error state and build the app.
- Log in to a store that's eligible for Blaze.
- If the campaign creation entry point is available on the Dashboard screen, select it and select a product if needed. Otherwise, navigate to the Products tab, select a published product and tap Promote with Blaze.
- On the campaign creation form, tap Confirm Details.
- Confirm that an alert is displayed with a button to retry on the payment screen.

## Screenshots

<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/b63a48b2-7eb6-42e9-a301-09ac8c3ac2ad" width=320 />


---
